src/app/notebook/notebook-mime-bundle-output.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { MarkdownWrapperService } from '../shared/markdown-wrapper/markdown-wrapper.service';
+import { escape, selectBestFromMimeBundle } from './notebook-helpers';
+import { MimeBundle, Output, OutputMetadata } from './notebook-types';
+
+@Component({
+  selector: 'app-notebook-mime-bundle-output',
+  templateUrl: './notebook-mime-bundle-output.component.html',
+})
+export class NotebookMimeBundleOutputComponent implements OnChanges {
+  @Input() output: Output;
+  html: string;
+
+  constructor(private markdownWrapperService: MarkdownWrapperService) {}
+
+  ngOnChanges(): void {
+    const mimeBundle = this.output?.data ?? {};
+    const metadataBundle = this.output?.metadata ?? {};
+    this.html = this.createHtmlFromBundles(mimeBundle, metadataBundle);
+  }
+
+  createHtmlFromBundles(mimeBundle: MimeBundle, metadataBundle: OutputMetadata): string {
```

Comment:
Should have `string | undefined` as the return type (returns undefined on line 42).